src/liballoc/arc.rs

```diff
@@ -54,16 +54,32 @@ const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 /// exception. If you need to mutate through an `Arc`, use [`Mutex`][mutex],
 /// [`RwLock`][rwlock], or one of the [`Atomic`][atomic] types.
 ///
-/// `Arc` uses atomic operations for reference counting, so `Arc`s can be
-/// sent between threads. In other words, `Arc<T>` implements [`Send`]
-/// as long as `T` implements [`Send`] and [`Sync`][sync]. The disadvantage is
-/// that atomic operations are more expensive than ordinary memory accesses.
-/// If you are not sharing reference-counted values between threads, consider
-/// using [`rc::Rc`][`Rc`] for lower overhead. [`Rc`] is a safe default, because
-/// the compiler will catch any attempt to send an [`Rc`] between threads.
-/// However, a library might choose `Arc` in order to give library consumers
+/// ## Thread Safety
+///
+/// Unlike [`Rc<T>`], `Arc<T>` uses atomic operations for its reference
+/// counting.  This means that it is thread-safe. The disadvantage is that
+/// atomic operations are more expensive than ordinary memory accesses.  If you
+/// are not sharing reference-counted values between threads, consider using
+/// [`Rc<T>`] for lower overhead. [`Rc<T>`] is a safe default, because the
+/// compiler will catch any attempt to send an [`Rc<T>`] between threads.
+/// However, a library might choose `Arc<T>` in order to give library consumers
 /// more flexibility.
 ///
+/// `Arc<T>` will implement [`Send`] and [`Sync`] as long as the `T` implements
+/// [`Send`] and [`Sync`]. This may be a bit counter-intuitive at first: after
+/// all, isn't the point of `Arc<T>` thread safety? The key is this: `Arc<T>`
+/// is an implementation of multiple ownership that is thread safe, but it
+/// doesn't add thread safety to its data. Consider `Arc<RefCell<T>>`.
+/// `RefCell<T>` isn't [`Send`], and if `Arc<T>` was always [`Send`],
+/// `Arc<RefCell<T>>` would be as well. But then we'd have a problem:
+/// `RefCell<T>` is not thread safe; it keeps track of the borrowing count
+/// using non-atomic operations. It works the same way with [`Sync`].
```

Comment:
`RefCell<T>` is `Send` when `T` is `Send`. :-) I think you could either swap `RefCell<T>` for `Rc<T>` and change the "borrowing count" to "reference count," or you could keep `RefCell<T>` and change `Send` to `Sync` (since `RefCell<T>` is never `Sync`) and then drop the last sentence.